zcash_primitives/src/legacy/keys.rs

```diff
@@ -111,11 +112,8 @@ impl AccountPubKey {
     ///
     /// [transparent-ovk]: https://zips.z.cash/zip-0316#deriving-internal-keys
     pub fn ovks_for_shielding(&self) -> (InternalOvk, ExternalOvk) {
-        let i_ovk = prf_expand_vec(
-            &self.0.chain_code,
-            &[&[0xd0], &self.0.public_key.serialize()],
-        );
-        let i_ovk = i_ovk.as_bytes();
+        let i_ovk = PrfExpand::TRANSPARENT_ZIP316_OVK
+            .with(&self.0.chain_code, &self.0.public_key.serialize());
```

Comment:
One minor complaint about the `with` macro: all the macro noise makes it somewhat difficult to understand the difference here, in terms of how the domain separators are included. It would be nice for the source documentation around the macro to give some examples of the code it generates.

Reply:
What the macro is generating is:
```rust
fn with(self, sk: &[u8], a: &[u8; A], b: &[u8; B] -> [u8; 64] {
    self.apply(sk, &[a, b])
}
```
i.e. `PrfExpand::with` is a type safety wrapper (that takes fixed-length inputs) around `PrfExpand::apply` (the latter allows you to use arbitrary-length inputs, so we don't expose it publicly).